ctags/main/entry.c

```diff
@@ -387,7 +387,7 @@ extern void makeTagEntry (const tagEntryInfo *const tag)
 {
 	Assert (tag->name != NULL);
 	if (tag->name [0] == '\0')
-		error (WARNING, "ignoring null tag in %s", vStringValue (File.name));
+		error (WARNING, "ignoring null tag in %s", getInputFileName ());
```

Comment:
The difference is only when #line directives are used - the "input" value is without line directives processed (which we want because we want the lines in source files) while "source" is when the directives are taken into account. But I think this happens only when some command-line switch is enabled and if not, both are the same.